### DISCUSSION TITLE
placeholderData and loading state

### DISCUSSION THREAD

**Discussion by bacarybruno** [TanStack/query#7944] [Q&A]
Heyo everyone 👋

I want to apply `placeholderData: keepPreviousData` in the default options of my query client. I want to do it to avoid hard loading states when data is available. But apparently, defining `placeholderData: keepPreviousData` means that we'll never have a hard loading state.

I still want to hard that hard loading state when the query is loading for the first time.
Or said differently, I want to apply `placeholderData: keepPreviousData` only after the first loading. Is it doable? 🙏

**Comment by TkDodo** [comment]
there should be a hard loading state when there is no data in the cache when the component mounts. Can you show an example?

**Reply by bacarybruno** [reply]
I cannot reproduce consistently. I'll try to setup a minimal repro and see what conditions triggers the issue I had
